synapse/util/caches/stream_change_cache.py

```diff
@@ -85,7 +85,7 @@ def has_entity_changed(self, entity: EntityType, stream_pos: int) -> bool:
         """Returns True if the entity may have been updated since stream_pos"""
         assert isinstance(stream_pos, int)
 
-        if stream_pos < self._earliest_known_stream_pos:
+        if stream_pos <= self._earliest_known_stream_pos:
```

Comment:
`has_entity_changed` is called from the following places:

* `AccountDataWorkerStore.get_updated_account_data_for_user`, which seems to be used to build account data in `/sync` responses.
* `DeviceInboxWorkerStore`:
	* `_get_device_messages`: feeds into to-device updates in `/sync` and app services.
	* `delete_messages_for_device`: used to pause `/sync` in some situation? (Unclear...?)
	* `get_new_device_msgs_for_remote`: used when calculating EDUs to send in federation transactions.
* `DeviceWorkerStore`:
	* `get_device_updates_by_remote`: used when calculating EDUs to send in federation transactions.
	* `get_users_whose_signatures_changed`: used in the device list entry for `/sync`.
* `EventPushActionsWorkerStore._get_notif_unread_count_for_user_room`, which is used to generate notification counts in `/sync` and to update `event_push_summary`.
* `PushRulesWorkerStore.have_push_rules_changed_for_user`, which is used to build push rules in `/sync` responses.
* `ReceiptsWorkerStore.get_linearized_receipts_for_room`, which is used build receipts data in `/initial_sync`.
* The `StreamWorkerStore` uses it in a few spots:
	* `get_rooms_that_changed`: feeds into the `/keys/changes` response.
	* `get_room_events_stream_for_room`: used to get the events that have changed in a room for `/sync` response.
	* `get_membership_changes_for_user`: feeds into the `/keys/changes` response. generating membership changes in `/sync` response, and whatever uses `RoomEventSource.get_new_events`.
* `TagsWorkerStore.get_updated_tags`: used to generate data in a `/sync` response.